### DISCUSSION TITLE
Should this be considered a publish failure?

### DISCUSSION THREAD

**Discussion by dmcweeney** [nrwl/nx#26010] [Questions] [answered]
I noticed that if the release version step does not work or if a build happens after the version the release publish will not upload the package with the following information...

Skipped package "xyz" from project "project-name" because v0.1.0 already exists in registry with tag "latest"

Should this be considered a failure as there is no way the publish pipeline can detect if the publish step fails?

From `packages\js\src\executors\release-publish\release-publish.impl.ts`

```
if (distTags[tag] === currentVersion) {
        console.warn(
          `Skipped ${packageTxt} because v${currentVersion} already exists in ${registry} with tag "${tag}"`
        );
        return {
          success: true,
        };
      }
```

**Comment by JamesHenry** [comment, marked as answer]
Having the publish step for a particular package be safe to run multiple times without error is an important characteristic because it means you can safely rerun publishing for any and all packages and not end up stuck and needing to manually figure out the right subset to run. A common case would be npm registry server issues that result in only some of your packages being published - no problem just rerun your release job and catch up.

I believe that should remain the default behaviour, but I think it would be fair to offer an option for the publish step to turn this into a hard error if you are confident that that is what you would always want. We would accept a community PR for that.

**Comment by dmcweeney** [comment]
@JamesHenry thanks for the answer - I was thinking there was solid reasoning behind it!

> publish step to turn this into a hard error if you are confident that that is what you would always want. We would accept a community PR for that.

I'm totally swamped for the next few weeks - I will try and get to it then...
